prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -301,83 +309,150 @@ class CheckFunction : Logging {
             responseBody.add("**** ${receiver.fullName}:  no transport defined.")
             return false
         }
+        responseBody.add("**** Receiver Status: ${receiver.customerStatus}")
+        val receiverConnected = when (receiver.transport) {
+            is SFTPTransportType -> testSftp(receiver.transport, receiver, sftpFile, responseBody)
+            is RESTTransportType -> testRest(receiver.transport, receiver, responseBody)
+            // todo add other types of transports as needed.
+            else -> {
+                responseBody.add(
+                    "**** ${receiver.fullName}: No test implemented for transport type ${receiver.transport.type}"
+                )
+                false
+            }
+        }
+        return if (receiverConnected) {
+            responseBody.add("**** ${receiver.fullName}: OK")
+        } else {
+            responseBody.add("**** ${receiver.fullName}: FAILED")
+            return false
+        }
+    }
+
+    /**
+     * Check if the REST transport can connect to the receiver
+     * and authenticate with the credentials in the vault
+     * Any normal return is success.  Any exception thrown is failure.
+     */
+    private fun testRest(
```

Comment:
I know `testRest` follows the naming convention in the file, but I kind of hate it uses `'test`` instead of `check` since tests often imply unit tests. (Nothing to do just thinking out loud).